src/main/java/com/wavefront/sdk/grpc/reporter/WavefrontGrpcReporter.java

```diff
@@ -0,0 +1,186 @@
+package com.wavefront.sdk.grpc.reporter;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import com.wavefront.internal.reporter.SdkReporter;
+import com.wavefront.internal.reporter.WavefrontInternalReporter;
+import com.wavefront.internal_reporter_java.io.dropwizard.metrics5.MetricName;
+import com.wavefront.sdk.common.WavefrontSender;
+import com.wavefront.sdk.common.application.ApplicationTags;
+import com.wavefront.sdk.common.application.HeartbeaterService;
+import com.wavefront.sdk.entities.metrics.WavefrontMetricSender;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Supplier;
+import java.util.logging.Logger;
+
+import javax.annotation.Nullable;
+
+import static com.wavefront.sdk.common.Constants.APPLICATION_TAG_KEY;
+import static com.wavefront.sdk.grpc.Constants.GRPC_CLIENT_COMPONENT;
+import static com.wavefront.sdk.grpc.Constants.GRPC_SERVER_COMPONENT;
+
+/**
+ * Wavefront reporter for your gRPC based application responsible for reporting metrics and
+ * histograms out of the box for you. Typically this is instantiated once per service and passed
+ * to grpc client/server tracers/interceptors.
+ *
+ * @author Srujan Narkedamalli (snarkedamall@wavefront.com).
+ */
+public class WavefrontGrpcReporter implements SdkReporter {
+  private static final Logger logger =
+      Logger.getLogger(WavefrontGrpcReporter.class.getCanonicalName());
+  private final int reportingIntervalSeconds;
+  private final WavefrontInternalReporter wfReporter;
+  private final WavefrontMetricSender wavefrontMetricSender;
```

Comment:
might want to keep everything consistent and go with wf prefix
for instance wfMetricSender